cpnest/proposal.py

```diff
@@ -5,11 +5,22 @@
 from abc import ABCMeta,abstractmethod
 import random
 from random import sample,gauss,randrange,uniform
-from scipy.interpolate import LSQUnivariateSpline
-from scipy.signal import savgol_filter
 from scipy.stats import multivariate_normal
 from scipy.special import logsumexp
 
+try:
+    from jax import grad, jit
+    FINITEDIFFERENCING = False
+except:
```

Comment:
Don't use a bare except. Use `except ImportError` if that is what you want to catch